api/apps/api/src/modules/scenario-specification/scenario-specification.ts

```diff
@@ -0,0 +1,264 @@
+    this.#currentCandidateSpec = specificationCandidate
+      ? Specification.from(specificationCandidate)
+      : undefined;
+  }
+
+  createCandidate(input: FeatureConfigInput[], draft: boolean) {
+    this.#currentCandidateSpec = Specification.new(input, draft);
+
+    const specSnapshot = this.#currentCandidateSpec.toSnapshot();
+    this.apply(
+      new CandidateSpecCreated(
+        specSnapshot.id,
+        specSnapshot.config.map((cfg) => ({
+          baseFeatureId: cfg.baseFeatureId,
+          againstFeatureId: cfg.againstFeatureId,
+          operation: cfg.operation,
+        })),
+      ),
+    );
+  }
+
+  /**
+   * consumes FeaturesFound
+   *
+   * may lead to activating Candidate Specification
+   * as long as all found features are already calculated
+   * and specification was marked as 'Create'
+   *
+   */
+  determineFeatures(features: FeaturesFound['spec']) {
+    this.#currentCandidateSpec?.determineFeatures(features);
+    this.publishCurrentCandidate();
+  }
+
+  /**
+   * may lead to activating the Candidate Spec
+   */
+  markFeaturesAsCalculated(features: string[]) {
+    if (this.#currentCandidateSpec?.markAsCalculated(features)) {
+      this.markCandidateAsActive();
+    }
+  }
+
+  getActiveSpecification = () => this.#currentActiveSpec?.toSnapshot();
+  getCandidateSpecification = () => this.#currentCandidateSpec?.toSnapshot();
+
+  private publishCurrentCandidate() {
+    if (!this.#currentCandidateSpec) {
+      return;
+    }
+    const outcome = this.#currentCandidateSpec.publish();
+
+    if (!outcome) {
+      return;
+    }
+
+    if (isLeft(outcome)) {
+      // if draft, its fine
+      // if created, we will wait anyway for all features
+      return;
+    }
+
+    if (outcome.right === SpecCanBePublished) {
+      this.apply(
+        new SpecificationPublished(
+          this.#currentCandidateSpec.id,
+          this.#currentCandidateSpec.relatedSubFeaturesRequiringCalculations(),
+        ),
+      );
+    }
+
+    if (outcome.right === SpecCanBeActivated) {
+      this.markCandidateAsActive();
+    }
+  }
+
+  private markCandidateAsActive() {
+    if (!this.#currentCandidateSpec) {
+      return;
+    }
+    if (this.#currentCandidateSpec.activate()) {
+      this.#currentActiveSpec = this.#currentCandidateSpec;
+      this.#currentCandidateSpec = undefined;
+      this.apply(new SpecificationActivated(this.#currentActiveSpec.id));
+    }
+  }
+}
+
+const DraftNotPublished = Symbol(`draft-not-published`);
+const SpecMissingFeatures = Symbol(`spec-missing-features`);
+const SpecCanBePublished = Symbol(`spec-can-be-published`);
+const SpecCanBeActivated = Symbol(`spec-can-be-activated`);
+
+type SpecPublishSuccess = typeof SpecCanBePublished | typeof SpecCanBeActivated;
+type SpecPublishFailure = typeof DraftNotPublished | typeof SpecMissingFeatures;
+
+class Specification {
+  private constructor(
+    public readonly id: string,
+    private configuration: FeatureConfig[] = [],
```

Comment:
```suggestion
    private configurations: FeatureConfig[] = [],
```

Reply:
@Dyostiq its still a single configuration (i.e. a group of related things)